### PR TITLE
Fix liquid bottom

### LINKED ISSUE #2674: Missing faces for liquids
There still are missing faces for liquids (probably related to #439):
screenshot_20150505_141135

To reproduce, build the following structure:
screenshot_20150505_141901
then look from the side:
screenshot_20150505_141909

### PR DESCRIPTION
**Pull request by numberZero**
Fixes #2674 
screenshot_20180129_202314

### CODE REVIEW

**Comment by Wuzzy2** [comment]
I have tested this. Works good. :+1:

File: src/content_mapblock.cpp [hunk 1]
```diff
@@ -385,9 +385,16 @@ void MapblockMeshGenerator::prepareLiquidNodeDrawing()
 	getSpecialTile(1, &tile_liquid);
 
 	MapNode ntop = data->m_vmanip.getNodeNoEx(blockpos_nodes + v3s16(p.X, p.Y + 1, p.Z));
+	MapNode nbottom = data->m_vmanip.getNodeNoEx(blockpos_nodes + v3s16(p.X, p.Y - 1, p.Z));
 	c_flowing = nodedef->getId(f->liquid_alternative_flowing);
 	c_source = nodedef->getId(f->liquid_alternative_source);
 	top_is_same_liquid = (ntop.getContent() == c_flowing) || (ntop.getContent() == c_source);
+	draw_bottom = (nbottom.getContent() != c_flowing) && (nbottom.getContent() != c_source);
+	if (draw_bottom && (nbottom.getContent() != CONTENT_AIR)) {
```
Comment (SmallJoker) [inline src/content_mapblock.cpp]: Also for `airlike` drawtypes
Superfluous parentheses

File: src/content_mapblock.cpp [hunk 2]
```diff
@@ -595,6 +602,24 @@ void MapblockMeshGenerator::drawLiquidTop()
 	collector->append(tile_liquid_top, vertices, 4, quad_indices, 6);
 }
 
+void MapblockMeshGenerator::drawLiquidBottom()
+{
+	video::S3DVertex vertices[4] = {
+		video::S3DVertex(-BS / 2,  -BS / 2, -BS / 2, 0, 0, 0, color_liquid_top, 0, 0),
+		video::S3DVertex( BS / 2,  -BS / 2, -BS / 2, 0, 0, 0, color_liquid_top, 1, 0),
+		video::S3DVertex( BS / 2,  -BS / 2,  BS / 2, 0, 0, 0, color_liquid_top, 1, 1),
```
Comment (SmallJoker) [inline src/content_mapblock.cpp]: Trivial double space in the 2nd argument

File: src/content_mapblock.h [hunk 3]
```diff
@@ -92,6 +92,7 @@ class MapblockMeshGenerator
 
 // liquid-specific
 	bool top_is_same_liquid;
+	bool draw_bottom;
```
Comment (SmallJoker) [inline src/content_mapblock.h]: Irrelevant for regular nodes -> `draw_liquid_bottom`

File: src/content_mapblock.cpp (hunk 1 above)
Comment (numberZero) [reply src/content_mapblock.cpp]: The check for air was redundant; removed it as that’s not really a common case.

File: src/content_mapblock.cpp (hunk 2 above)
Comment (numberZero) [reply src/content_mapblock.cpp]: Fixed.

File: src/content_mapblock.h (hunk 3 above)
Comment (numberZero) [reply src/content_mapblock.h]: Renamed.

**Pull request merged**